repos/spack_repo/builtin/packages/silo/package.py

```diff
@@ -242,14 +295,27 @@ def configure_args(self):
                 ]
             )
 
-        if "+mpi" in spec:
-            config_args.append("CC=%s" % spec["mpi"].mpicc)
-            config_args.append("CXX=%s" % spec["mpi"].mpicxx)
-            config_args.append("FC=%s" % spec["mpi"].mpifc)
-
         return config_args
 
     @property
     def libs(self):
         shared = "+shared" in self.spec
         return find_libraries("libsilo*", root=self.prefix, shared=shared, recursive=True)
+
+
+class CMakeBuilder(cmake.CMakeBuilder):
+    def cmake_args(self):
+        args = [
+            self.define("SILO_ENABLE_JSON", False),
+            self.define("SILO_ENABLE_ASAN", False),
+            self.define("SILO_ENABLE_SILOCK", True),
+            self.define("SILO_ENABLE_BROWSER", True),
+            self.define("SILO_ENABLE_INSTALL_LITE_HEADERS", True),
```

Comment:
Ok,just realized, I am not handling `SILO_BUILD_FOR_BSD` here yet. That is an oversight. Also, `SILO_ENABLE_ASAN` is not a Silo CMake variable. I need to fix these

Reply:
@patrickb314 what is the preferred way of dealing with CMake variables here that already default to the desired settings in the actual package? Options are to not reference them at all here or to explicitly set them here as well even though they simply repeat the package's default settings for those variables.